Editor/AGXUnityEditor/BrickUnity/BrickPrefabImporter.cs

```diff
@@ -137,19 +136,99 @@ public GameObject HandleNode(B_Node b_node, GameObject go, GameObject go_parent
+            {
+              go_rb_external = c_uuid.gameObject;
+              bodyFound = true;
+            }
+
+            // Try to find external body by name
+            bool testRb(AGXUnity.RigidBody rb) => rb.name == extRef;
+            var c_rb = go_external.GetComponentsInChildren<AGXUnity.RigidBody>().SingleOrDefault(testRb);
+            if (c_rb != default)
+            {
+              go_rb_external = c_rb.gameObject;
+              bodyFound = true;
+            }
+
+            if (!bodyFound)
+              throw new AGXUnity.Exception($"No external body with name or UUID \"{extRef}\" was found.");
+
+            Object.DestroyImmediate(go);
+            go = go_rb_external;
+          }
           bodyDict.Add(b_body, go);
           break;
         case B_RbAttachment b_rbAttachment:
-          attachmentDict.Add(b_rbAttachment, go);
           break;
         case B_Visual.Shape b_visualShape:
           go = HandleVisuals(go, b_visualShape);
           break;
+        case B_Component b_component:
+          // Find the full path of the external file (if it is relative)
+          if (!b_component._externalFilepathIsDefault)
+          {
+            var externalFilepath = b_component.ExternalFilepath;
+            var componentFilePath = b_component._ModelValue.File.Filepath;
+            var componentFolderPath = Path.GetDirectoryName(componentFilePath);
+            var fullPaths = new List<string>
+            {
+              Path.Combine(componentFolderPath, externalFilepath),
+              Path.GetFullPath(externalFilepath),
+              Path.Combine(System.Environment.GetEnvironmentVariable("BRICK_DIR"), externalFilepath)
+            };
+
+            var fullPath = fullPaths.FirstOrDefault(path => File.Exists(path));
+            if (fullPath == default)
+            {
+              Debug.LogError("Paths searched:\n  " + string.Join("\n  ", fullPaths));
+              throw new FileNotFoundException("External file not found", externalFilepath);
+            }
+
+            // Copy the AGX file to the Data directory (under Assets), then use AGXUnity to import it
+            var ext = Path.GetExtension(fullPath);
+            if (ext == ".agx" || ext == ".aagx")
+            {
+              var filename = Path.GetFileName(fullPath);
+              var dataFilepath = Path.Combine(this.DataDirectoryPath, filename);
+              GetOrCreateDataDirectory();
+              File.Copy(fullPath, dataFilepath, true);
+              var oldName = go.name;
+              Object.DestroyImmediate(go);
+              var agxPrefab = AGXFileImporter.Import(dataFilepath);
+              go = Object.Instantiate(agxPrefab);
+              go.name = oldName + " (external)";
+              go_external = go;
+            }
+            else
+            {
+              throw new AGXUnity.Exception($"Brick could not load external file {fullPath}. Unknown file extension: {ext}");
+            }
+          }
+          break;
```

Comment:
There is a lot of code in this switch-statement now. Maybe some parts of the B_RigidBody and B_Component-cases can be moved to its own method, just to make things look more clean?

Reply:
Fixed in ae47429061ffa3a4611f61de6dc423fe483e25ff